drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -32,6 +33,7 @@ struct mcux_flexcomm_config {
 	void (*irq_config_func)(const struct device *dev);
 	uint32_t bitrate;
 	const struct pinctrl_dev_config *pincfg;
+	const struct reset_dt_spec reset;
```

Comment:
Does this need to be conditional? It would seem the flexcomm driver is used by other families as well.

Reply:
nope, trust me, this issue of IP shared by families was one I was well aware of to implement this PR, all the platforms with flexcomm are addressed in this PR

also, I am planning to convert all the NXP IP and platform eventually, this PR is just the first one introducing the concept to the nxp vendor code bubble within zephyr, and it is both the minimum and the maximum of code changes I found that I could do to completely remove the reset code from the LPC soc.c files